### DISCUSSION TITLE
Custom field on product options breaks GQL schema

### DISCUSSION THREAD

**Discussion by VojtaSim** [swellstores/community#547] [Help] [answered]
Hello,

I've added a custom field to product option (Admin field location = "Option edit") and it broke the GraphQL schema and I can't query product with options since I get error: `Cannot query field \"id\" on type \"SwellProductOption\".`

My custom fields for product model are:
<img width="1241" alt="image" src="https://github.com/swellstores/community/assets/4440970/47cccd7e-7413-4acd-871b-8418234d3ae7">

The first one is the one that breaks it, 2nd one works fine.

This is what GQL schema looks like after adding the field.
<img width="669" alt="image" src="https://github.com/swellstores/community/assets/4440970/efd6b676-ef2e-4b08-a709-279f5acb5db7">

```
query ProductById {
    productById(id: "...") {
        id
        name
        slug
        price
        sale
        origPrice
        currency
        options {
            id
            name
            price
            description
            variant
            required
            parentId
            parentValueIds
            pricePercentIncrease
            values {
                id
                name
                price
                description
                pricePercentIncrease
            }
        }
    }
}
```

This is a second time I'm giving Swell a shot after previous unpleasant experience but seems like not much has changed and it's bugged from ground up.

**Comment by sags95** [comment, marked as answer]
I'm not quite able to replicate the error, are the product options being generated correctly? Sometimes it does take a bit of time for the schema to update as well (we cache it).


<img width="549" alt="Screenshot 2024-06-13 at 10 39 34 AM" src="https://github.com/swellstores/community/assets/9522171/7211a2a1-f8ac-4da7-95b9-e10b4fc20e87">
<img width="1021" alt="Screenshot 2024-06-13 at 10 38 35 AM" src="https://github.com/swellstores/community/assets/9522171/16129a37-4a74-4f54-8e7c-f1de0fa9059d">
<img width="1000" alt="Screenshot 2024-06-13 at 10 38 26 AM" src="https://github.com/swellstores/community/assets/9522171/700de58e-7549-4400-b8cd-f09dea3b302f">

**Reply by VojtaSim** [reply]
Try to name it the same, `options.price_increase` and `options.values.price_increase`. It started working when I renamed it.

Worth to mention when I created the fields it took couple of sec. to show up in schema but it was working. Then after another few minutes it stopped (discovered it because app threw an error) and schema looked as I posted it above.

**Reply by sags95** [reply]
With the same field I'm still not able to reproduce unfortunately. Have you tried with another field?


<img width="579" alt="Screenshot 2024-06-17 at 11 40 07 AM" src="https://github.com/swellstores/community/assets/9522171/43cdb1d9-38f0-4509-9041-9693d4694482">
